packages/grafana-flamegraph/src/FlameGraphContainer.test.tsx

```diff
@@ -87,4 +91,26 @@ describe('FlameGraphContainer', () => {
 
     expect(screen.queryByTestId(/Both/)).toBeNull();
   });
+
+  it('should filter table items based on search input', async () => {
+    // Render the FlameGraphContainer with necessary props
+    render(<FlameGraphContainerWithProps />);
+
+    // Checking for presence of this function before filter
+    const textBeforeFilter = 'net/http.HandlerFunc.ServeHTTP';
+    const rowsBeforeFilter = screen.queryAllByText(textBeforeFilter);
+    expect(rowsBeforeFilter.length).toBeGreaterThan(0);
+
+    // Apply the filter
+    const searchInput = await screen.getByPlaceholderText('Search...');
+    await userEvent.type(searchInput, textBeforeFilter);
+
+    // Verify that the table displays filtered item
+    const filteredRows = screen.queryAllByText(textBeforeFilter);
+    expect(filteredRows.length).toBeGreaterThan(0);
+
+    // Verify that rows with some other text are not shown after filter
+    const rowsAfterFilter = screen.queryAllByText('compress/gzip.(*Writer).Write');
+    expect(rowsAfterFilter.length).toBe(0);
```

Comment:
We would be more likely to catch an issue with the filtering if we also check this value before we apply the search filter to make sure it exists. That way, after we apply the search filter we expect it not to exist but we know that it did previously exist.